File: code_translation/exp1_code_translation_security/translated_solutions_claude/Question4/Java/1038.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class DatabaseOperations {
    
    public void insertion(String name, int age) {
        String url = "jdbc:mysql://localhost:3306/your_database_name";
        String username = "your_username";
        String password = "your_password";
        
        try {
            // Create connection
            Connection con = DriverManager.getConnection(url, username, password);
            
            // SQL query with placeholders for parameters
            String sql = "INSERT INTO STUDENTS (NAME, AGE) VALUES (?, ?)";
            
            // Create prepared statement
            PreparedStatement pstmt = con.prepareStatement(sql);
            pstmt.setString(1, name);
            pstmt.setInt(2, age);
            
            // Execute the query
            int rowsAffected = pstmt.executeUpdate();
            
            if (rowsAffected > 0) {
                System.out.println("Record inserted successfully!");
            }
            
            // Close resources
            pstmt.close();
            con.close();
            
        } catch (SQLException e) {
            System.err.println("Database error: " + e.getMessage());
            e.printStackTrace();
        }
    }
}

File: code_translation/exp1_code_translation_security/translated_solutions_gpt-4o/Question4/Rust/1038.rs
use mysql::*;
use mysql::prelude::*;

fn insertion(name: &str, age: u32) -> Result<()> {
    // Create a connection to the MySQL database
    let url = "mysql://username:password@localhost:3306/database_name"; // Replace with your actual database credentials
    let pool = Pool::new(url)?;
    let mut conn = pool.get_conn()?;

    // SQL query to insert data into the STUDENTS table
    let sql = "INSERT INTO STUDENTS (NAME, AGE) VALUES (:name, :age)";

    // Execute the query with the provided parameters
    conn.exec_drop(sql, params! {
        "name" => name,
        "age" => age,
    })?;

    Ok(())
}